services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceParentHelixAdmin.java

```diff
@@ -2413,50 +2413,89 @@ public void updateStore(String clusterName, String storeName, UpdateStoreQueryPa
+       * 3. The store is not a system store and incremental push is not already enabled.
+       */
+      if (!currStore.isSystemStore() && !setStore.incrementalPushEnabled && setStore.activeActiveReplicationEnabled
+          && updatedHybridStoreConfig != null
+          && clusterConfig.enabledIncrementalPushForHybridActiveActiveUserStores()) {
+        setStore.incrementalPushEnabled = true;
+        updatedConfigsList.add(INCREMENTAL_PUSH_ENABLED);
       }
+
+      /*
+       * If active-active replication mode is being enabled, and hybrid data replication policy is not set,
+       * and the store is not a system store, then set data replication policy to ACTIVE_ACTIVE DRP.
+       */
+      if (!currStore.isSystemStore() && setStore.activeActiveReplicationEnabled
+          && !hybridDataReplicationPolicy.isPresent() && updatedHybridStoreConfig != null
+          && updatedHybridStoreConfig.getDataReplicationPolicy() != DataReplicationPolicy.ACTIVE_ACTIVE) {
+        LOGGER.info(
+            "Data replication policy was not explicitly set when enabling active-active replication mode for store: {}."
+                + ". Setting it to ACTIVE_ACTIVE DRP.",
+            storeName);
+        updatedHybridStoreConfig.setDataReplicationPolicy(DataReplicationPolicy.ACTIVE_ACTIVE);
+        updatedConfigsList.add(DATA_REPLICATION_POLICY);
+      }
+
       // When turning off hybrid store, we will also turn off A/A store config.
       if (storeBeingConvertedToBatch && setStore.activeActiveReplicationEnabled) {
         setStore.activeActiveReplicationEnabled = false;
         updatedConfigsList.add(ACTIVE_ACTIVE_REPLICATION_ENABLED);
       }
 
-      // Update incremental push config.
-      setStore.incrementalPushEnabled =
-          incrementalPushEnabled.map(addToUpdatedConfigList(updatedConfigsList, INCREMENTAL_PUSH_ENABLED))
-              .orElseGet(currStore::isIncrementalPushEnabled);
-      // Enable incremental push automatically when batch user store being converted to hybrid store and active-active
-      // replication is enabled or being and the cluster config allows it.
-      if (!setStore.incrementalPushEnabled && !currStore.isSystemStore() && storeBeingConvertedToHybrid
-          && setStore.activeActiveReplicationEnabled
-          && clusterConfig.enabledIncrementalPushForHybridActiveActiveUserStores()) {
-        setStore.incrementalPushEnabled = true;
-        updatedConfigsList.add(INCREMENTAL_PUSH_ENABLED);
-      }
       // When turning off hybrid store, we will also turn off incremental store config.
       if (storeBeingConvertedToBatch && setStore.incrementalPushEnabled) {
         setStore.incrementalPushEnabled = false;
         updatedConfigsList.add(INCREMENTAL_PUSH_ENABLED);
       }
 
+      /*
+       * If active-active replication mode is being disabled, and store is still a hybrid store,
+       * the store is not a system store, then
```

Comment:
Do we need to check system store in this case?

Reply:
Yes we do and it's part of the condition

Reply:
I mean whether we should remove the system store check; I think if we are disabling AA for system stores, we should update the DRP config too.

Reply:
We can but IDK if we should 😄